fpdf/drawing.py

```diff
@@ -0,0 +1,1023 @@
+        return (
+            f"{_render_number(self.a)} {_render_number(self.b)} "
+            f"{_render_number(self.c)} {_render_number(self.d)} "
+            f"{_render_number(self.e)} {_render_number(self.f)} cm",
+            last_point,
+        )
+
+    def __str__(self):
+        return (
+            f"transform: ["
+            f"{_render_number(self.a)} {_render_number(self.b)} 0; "
+            f"{_render_number(self.c)} {_render_number(self.d)} 0; "
+            f"{_render_number(self.e)} {_render_number(self.f)} 1]"
+        )
+
+
+class StrokeWidth:
+    def __init__(self, width):
+        self.width = width
+
+    def render(self, last_point):
+        return (f"{_render_number(self.width)} w", last_point)
+
+    def __str__(self):
+        return f"stroke width: {_render_number(self.width)}"
+
+
+class StrokeCapStyle(enum.IntEnum):
+    butt = 0
+    round = 1
+    square = 2
+
+    @classmethod
+    def coerce(cls, value):
+        if isinstance(value, cls):
+            return value
+        elif isinstance(value, str):
+            return cls[value.lower()]
+        elif isinstance(value, int):
+            return cls(value)
+        else:
+            raise TypeError(f"{value} cannot convert to a {cls.__name__}")
+
+    def render(self, last_point):
+        return (f"{self.value} J", last_point)
+
+    def __str__(self):
+        return f"stroke cap style: {self.name}"
+
+
+class StrokeJoinStyle(enum.IntEnum):
+    miter = 0
+    round = 1
+    bevel = 2
+
+    @classmethod
+    def coerce(cls, value):
+        if isinstance(value, cls):
+            return value
+        elif isinstance(value, str):
+            return cls[value.lower()]
+        elif isinstance(value, int):
+            return cls(value)
+        else:
+            raise TypeError(f"{value} cannot convert to a {cls.__name__}")
+
+    def render(self, last_point):
+        return (f"{self.value} j", last_point)
+
+    def __str__(self):
+        return f"stroke join style: {self.name}"
+
+
+class StrokeMiterLimit:
+    def __init__(self, limit):
+        self.limit = limit
+
+    def render(self, last_point):
+        # limit is a ratio so it does not need to be scaled
+        return (f"{_render_number(self.limit)} M", last_point)
+
+    def __str__(self):
+        return f"stroke miter limit: {_render_number(self.limit)}"
+
+
+class StrokeDashPattern:
+    def __init__(self, pattern, phase):
+        if isinstance(pattern, (int, float)):
+            pattern = (pattern,)
+
+        if len(pattern) > 2:
+            raise ValueError(f"dash pattern must have 2 or fewer entries")
+        self.pattern = pattern
+        self.phase = phase
+
+    def render(self, last_point):
+        return (
+            f"[{' '.join(_render_number(pat) for pat in self.pattern)}] {_render_number(self.phase)} d",
+            last_point,
+        )
+
+    def __str__(self):
+        return f"stroke dash: pattern={self.pattern} phase={self.phase}"
+
+
+class FillColor:
```

Comment:
Maybe this class should inherit from `Color`?
Same remark for `StrokeColor`

Reply:
Hmmm. `FillColor` and `StrokeColor` are mostly just shell classes that represent a specific style attribute of a graphics context, whereas the `Color` is intended to represent an arbitrary color specification by a set of values.

The way that the style management currently works is that each style item is represented by a unique class so it can be stored in a dictionary with its class as the key. Inheriting from `Color` won't help for that purpose, so maybe the intent of these classes can be improved by naming them `FillColorStyle` and `StrokeColorStyle`.

Reply:
The style handling has been completely rewritten and doesn't use these classes any more, so I think this is resolved.